zk/conn.go

```diff
@@ -964,6 +961,25 @@ func (c *Conn) queueRequest(opcode int32, req interface{}, res interface{}, recv
 	return rq.recvChan
 }
 
+func (c *Conn) queueRequestWithTimeout(opcode int32, req interface{}, res interface{}, recvFunc func(*request, *responseHeader, error), timeout time.Duration) <-chan response {
```

Comment:
consider: we could use a context to do this cancelation:

```
func (c *Conn) Close() {
  close(c.shouldQuit)
   ctx, cancel := context.WithTimeout(context.Backgroud(), time.Second)
  defer cancel()
  if err := queueRequestWithTimeout(ctx); err != nil {
    c.log("failed to close connection: %v", err)
  }
}

func (c *Conn) queueRequestWithTimeout(ctx context.Context) error {
      rq := &request{
		xid:        c.nextXid(),
		opcode:     opcode,
		pkt:        req,
		recvStruct: res,
		recvChan:   make(chan response, 1),
		recvFunc:   nil, // maybe pass into method for tests
	}
     select {
	case <-ctx.Done():
		rq.recvChan <- response{err: ErrConnectionClosed}
                return ctx.Err()
         case <-c.sendChan <- rq:
     }
    return nil
}
```

Reply:
Thanks for the review.  I added the new function to provide a capability to send a request with a timeout while keep same functionalities queueRequest() offers.  I like your idea of using a context instead though.  Updating the code.